File: Data Structures & Algorithms/search-2d-matrix/submission-0.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {

        int l = 0;
        int n = matrix[0].size();
        int r = (matrix.size() * matrix[0].size()) - 1;
        int mid;

        while(l <= r){
            mid = (l+r)/2;
            if(matrix[mid/n][mid%n] == target){
                return true;
            } else if (matrix[mid/n][mid % n] < target){
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        
        return false;
    }
};

File: Data Structures & Algorithms/car-fleet/submission-0.py
class Solution:

    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        cars = sorted(zip(position, speed), reverse=True)  # closest to target first
        stack = []  # times of distinct fleets, front-most fleet at bottom

        for pos, spd in cars:
            time = (target - pos) / spd
            if not stack or time > stack[-1]:
                stack.append(time)
            # else: this car catches up to the fleet ahead, no new fleet

        return len(stack)